crates/biome_js_analyze/src/lint/nursery/use_throw_new_error.rs

```diff
@@ -0,0 +1,180 @@
+use std::ops::Not;
+
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsExpression, JsCallExpression, JsNewExpression, JsParenthesizedExpression, JsSyntaxKind, T,
+};
+use biome_rowan::{AstNode, BatchMutationExt, TriviaPieceKind};
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Require new when throwing an error
+    ///
+    /// While it's possible to create a new error without using the new keyword, it's better to be explicit.
+    ///
```

Comment:
We should add an explanation of how the rule works, i.e. that all names that ends with `Error` and are capitalized are considered.